src/System.CommandLine/Command.cs

```diff
@@ -170,6 +117,56 @@ private protected override void AddSymbol(Symbol symbol)
         /// </summary>
         public IEnumerator<Symbol> GetEnumerator() => Children.OfType<Symbol>().GetEnumerator();
 
+        /// <summary>
+        /// Validates all symbols including the child hierarchy.
+        /// </summary>
+        /// <remarks>Due to the performance impact of this method, it's recommended to create
+        /// a Unit Test that calls this method to verify the RootCommand of every application.</remarks>
+        public void Validate()
+        {
+            for (int i = 0; i < Children.Count; i++)
+            {
+                for (int j = 1; j < Children.Count; j++)
+                {
+                    if (Children[j] is IdentifierSymbol identifierSymbol)
+                    {
+                        foreach (string alias in identifierSymbol.Aliases)
+                        {
+                            if (Children[i].Matches(alias))
+                            {
+                                throw new ArgumentException($"Alias '{alias}' is already in use.");
+                            }
+                        }
+
+                        if (identifierSymbol is Command command)
+                        {
+                            if (ReferenceEquals(this, command))
+                            {
+                                throw new ArgumentException("Parent can't be it's own child.");
```

Comment:
I would suggest a different exception type.

Also, minor correction: 

```suggestion
                                throw new ArgumentException("Parent can't be its own child.");
```